src/xmipp/libraries/data/monogenic.h

```diff
@@ -0,0 +1,172 @@
+/***************************************************************************
+ *
+ * Authors:     Jose Luis Vilas (jlvilas@cnb.csic.es)
+ *
+ * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+ * 02111-1307  USA
+ *
+ *  All comments concerning this program package may be sent to the
+ *  e-mail address 'xmipp@cnb.csic.es'
+ ***************************************************************************/
+#ifndef _MONOGENIC_HH
+#define _MONOGENIC_HH
+
+#include <stdio.h>
+#include <iostream>
+#include <math.h>
+#include <stdlib.h>
+#include <core/metadata.h>
+#include <core/xmipp_image.h>
+#include <data/sampling.h>
+#include <core/xmipp_fft.h>
+#include <core/xmipp_fftw.h>
+#include <math.h>
+#include <limits>
+#include <complex>
+#include <data/fourier_filter.h>
+#include <data/filters.h>
+//@{
+/** Routines for working with monogenic signals
+*/
+
+class Monogenic
+{
+public:
+	// PROTEINRADIUSVOLUMEANDSHELLSTATISTICS: This function takes as input a "mask" and returns
+	// the radius and the volumen "vol" of the protein. The radius is defines has the distance
+	// from the center of the cube to the farthest point of the protein measured from the center.
+	// The volumen "vol" represent the number of voxels of the mask.
+	void proteinRadiusVolumeAndShellStatistics(MultidimArray<int> &mask, double &radius,
+			long &vol, MultidimArray<double> &radMap);
+
+
+	// FINDCLIFFVALUE: This function determines the radius of noise, "radiuslimit". It means given
+	// a map, "inputmap", the radius measured from the origen for which the map is masked with a
+	// spherical mask is detected. Outside of this sphere there is no noise. Once the "mask" is
+	// set to -1 for all voxels with radius greater than "radiuslimit". The parameter "rsmooth"
```

Comment:
'once the "mask" ...' sentence does not make sense

Reply:
fixed